include/mega/node.h

```diff
@@ -67,8 +69,16 @@ struct MEGA_API NewNode : public NodeCore
 
     bool added;
 
+    MEGA_DISABLE_COPY(NewNode)
+
+    NewNode(NewNode&& o);
+    NewNode& operator=(NewNode&& o);
+
     NewNode();
     ~NewNode();
+
+private:
+    void swap(NewNode&& o);
```

Comment:
swap should not be needed - I see it's used from the move constructor, but there's no need to copy this object's members back to the source object